File: LaGranjaSA/src/main/java/com/example/LaGranjaSA/modelo/Porcino.java
package com.example.LaGranjaSA.modelo;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.Period;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Table(name = "porcino")
public class Porcino {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id_porcino;
    private LocalDate fecha_nacimiento;
    private float peso;
    @ManyToOne
    @JoinColumn(name = "fk_id_cliente")
    private Cliente cliente;

    @ManyToOne
    @JoinColumn(name = "fk_id_raza")
    private Raza raza;
    private int edad;
    public int calcularEdad() {
        if (fecha_nacimiento == null) {
            return 0;

        }
        return Period.between(fecha_nacimiento, LocalDate.now()).getDays();
    }

}

File: LaGranjaSAFront/LaGranjaFront/src/app/model/Porcino.ts
import { Cliente } from "./Cliente";
import { Raza } from "./Raza";

export class Porcino {
    public id_porcino: number;
    public fecha_nacimiento: Date;
    public peso: number;
    public edad : number;
    public raza: Raza;
    public cliente: Cliente;
    
    constructor(fecha_nacimiento: Date, peso: number) {
        this.fecha_nacimiento = fecha_nacimiento; 
        this.peso = peso; 
    }


}